### PR TITLE
Report fix

### LINKED ISSUE #429: Fix FinishFlightDetails logic to have add report

### PR DESCRIPTION
**Pull request by hacknstack**
closes #429,
Since we still have the Admin graph bug in stats screen, the see report shows the report of everyone always.

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/screens/crewpilot/FlightDetailsScreen.kt [hunk 1]
```diff
@@ -35,7 +36,23 @@ fun CrewFlightDetailScreen(
           }
           is FinishedFlight -> {
             FinishedFlightDetailBottom(
-                reportClick = { navController.navigate(Route.REPORT + "/${flightId}") },
+                reportClick = {
+                  val reportDone =
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/screens/crewpilot/FlightDetailsScreen.kt:40]: would be cleaner to have the whole logic in the ViewModel. Here you could add a function to the VM like for example:
`reportDoneFor(flightid, userId): Boolean` that can then simply be accessed through the VM

File: app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt [hunk 2]
```diff
@@ -37,8 +38,10 @@ fun ReportDetailsScreen(
     val display = (isAdmin && reportIds!!.isNotEmpty()) || reportIds!!.any { (it.author == userId) }
     Column() {
       CustomTopAppBar(navController, "Report")
+      reportIds!!.forEach { report -> Log.d("report", report.author) }
       if (display) {
-        ReportDetail(users, reportIds, isAdmin, userId)
+        val reportLists = if (isAdmin) reportIds else reportIds!!.filter { (it.author == userId) }
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt:43]: same as above. Ideally, the VM (which know the user) will return the relevant reportIds

**Review by RAFInery — APPROVED** [review]
I let few comments but apart from that looks fine

File: app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt (hunk 2 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt:43]: Big iq comment ty

File: app/src/main/java/ch/epfl/skysync/screens/crewpilot/FlightDetailsScreen.kt (hunk 1 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/screens/crewpilot/FlightDetailsScreen.kt]: 👍

**Pull request merged**